Server/controllers/authController.js

```diff
@@ -234,11 +234,37 @@ const resetPasswordController = async (req, res, next) => {
     next(error);
   }
 };
+
+const deleteUserController = async (req, res, next) => {
+  try {
+    // Validate user
+    await editUserParamValidation.validateAsync(req.params);
+
+    // Check if the user exists
+    const user = await req.db.getUserById(req.params.userId);
+    if (!user) {
+      throw new Error(errorMessages.DB_USER_NOT_FOUND);
+    }
+
+    // Delete the user
+    await req.db.deleteUser(req.params.userId);
```

Comment:
When a user is deleted, their associated monitors should be deleted as well as all checks and alerts associated with their monitor.

The delete should cascade to all documents the user owns.

Reply:
@ajhollid Please check the complementary modifications